course-project/Backend/src/cmd/rest-api/lib/proxy_test.go

```diff
@@ -47,3 +48,41 @@ func TestGetCurrentWeatherFromAPI(t *testing.T) {
 		})
 	}
 }
+
+func TestParseJSONFromApi(t *testing.T) {
+	temp := make([]map[string]interface{}, 0)
+	b, _ := json.Marshal(temp)
+	t1 := make(map[string]interface{})
+	l1 := make(map[string]interface{})
+	l1["temp"] = 17.0
+	t1["main"] = l1
+	t1["name"] = "test"
+	temp = append(temp, t1)
+	var ff Forecast
+	ff.List = temp
+	b1, _ := json.Marshal(ff)
+	type args struct {
+		body []byte
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    Weather
+		wantErr bool
```

Comment:
`wantErr` needs to be of type `error`. you need to check for the value of error. Need to fix in other tests as well